### ISSUE TITLE
dbus "watch barrier" unit test fails often

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#15095]
In most of our CI runs, the `watch barrier` unit test fails [like this](https://github.com/cockpit-project/cockpit/pull/15088/checks?check_run_id=1649726119#step:4:3120):
```
> log: not ok 120 - correct data, expected: {
#   "FinallyNormalName": "There aint no place like home",
#   "ReadonlyProperty": "blah",
#   "aay": [],
#   "ag": [],
#   "ao": [],
#   "as": [],
#   "ay": "QUJDYWJjAA==",
#   "b": false,
#   "d": 43,
#   "g": "",
#   "i": 0,
#   "n": 0,
#   "o": "/",
#   "q": 0,
#   "s": "",
#   "t": 0,
#   "u": 0,
#   "x": 0,
#   "y": 42
# }, got: {}, test: watch barrier, module: , source: at Function.<anonymous> (http://localhost:45985/dist/base1/test-dbus-address.js:1:18049)
#     at http://localhost:45985/dist/base1/cockpit.js:1:13301
#     at http://localhost:45985/dist/base1/cockpit.js:1:13411
#     at A (http://localhost:45985/dist/base1/cockpit.js:1:12825)
```

I got this once or twice on unrelated local builds on my machine, so it is *not* specific to GitHub nor our tasks container. It just so happens that the timing on GitHub is "right" for this to appear. But we have had this for a looong time, we often saw it on semaphore as well. But now it became really  bad.

I am using this issue to collect some debugging notes. I already tried in vain to reproduce this interactively in GitHub ssh session, and in PR #14985 I hoped to have fixed it, but unfortunately not.

So I am now throwing test builds [against my fork](https://github.com/martinpitt/cockpit/actions) to drill down into this.

**Comment by martinpitt** [comment]
* Pretty much any attempt to reduce this problem makes it go away. E.g. only running that single test, or only running the D-Bus tests. There needs to be a well-defined amount of noise going on.
* I added some extensive logging and enabled cockpit debugging, and it [succeeded](https://github.com/martinpitt/cockpit/runs/1650103726?check_suite_focus=true). I am now trying to re-run it several times, it may just have been bad/good luck.

**Comment by martinpitt** [comment]
Argh, I feel rather embarrassed now -- it was me who broke this :crying_cat_face: . Fix sent in PR #15096
